### ISSUE TITLE
Error 500 when submitting database info

### ISSUE THREAD

**Issue by lcsqlpete** [jan-vandenberg/cruddiy#131]
Right after submitting the database info during setup, I get an error 500 message from my server.  Looking at the logs, I see that this was caused by "PHP Fatal error:  Uncaught mysqli_sql_exception: Invalid character set was provided" on line 41 of the config.php file in the cruddiy-master/code/myapp/config.ini

**Comment by germain-italic** [comment]
Hi, there's no config.ini file in Cruddiy.
This looks like a local configuration problem.
What's in that file?

**Comment by lcsqlpete** [comment]
Sorry, I meant config.php

**Comment by germain-italic** [comment]
@lcsqlpete your database uses the `utf8mb3` charset which is [deprecated](https://dev.mysql.com/doc/refman/8.3/en/charset-unicode-utf8mb3.html) and has been removed from specific distributed packages.

You must use `utf8mb4` to correctly support Unicode characters.

That being said, I've published a patch that will fallback the generator's connexion to `utf8mb4` if an unrecognized charset is detected.

The current charset for the connection is now displayed after config submission:

![](https://files.italic.fr/chrome_MwBDlm8hZz.png)

This patch is in the master branch and hasn't been released yet, so [download the ZIP file](https://github.com/jan-vandenberg/cruddiy/archive/refs/heads/master.zip) from the project homepage instead of the Releases page.

Also, **change your MySQL server password immediately!**
